examples/multigpu/graphbolt/node_classification.py

```diff
@@ -0,0 +1,444 @@
+        return hidden_x
+
+
+def create_dataloader(
+    args,
+    graph,
+    features,
+    itemset,
+    device,
+    drop_last=False,
+    shuffle=True,
+    drop_uneven_inputs=False,
+):
+    ############################################################################
+    # [HIGHLIGHT]
+    # Get a GraphBolt dataloader for node classification tasks with multi-gpu
+    # distributed training. DistributedItemSampler instead of ItemSampler should
+    # be used.
+    ############################################################################
+
+    ############################################################################
+    # [Step-1]:
+    # gb.DistributedItemSampler()
+    # [Input]:
+    # 'item_set': The current dataset. (e.g. `train_set` or `valid_set`)
+    # 'batch_size': Specifies the number of samples to be processed together,
+    # referred to as a 'mini-batch'. (The term 'mini-batch' is used here to
+    # indicate a subset of the entire dataset that is processed together. This
+    # is in contrast to processing the entire dataset, known as a 'full batch'.)
+    # 'drop_last': Determines whether the last non-full minibatch should be
+    # dropped.
+    # 'shuffle': Determines if the items should be shuffled.
+    # 'num_replicas: Specifies the number of replicas.
+    # 'drop_uneven_inputs': Determines whether the numbers of minibatches on all
+    # ranks should be kept the same by dropping uneven minibatches.
+    # [Output]:
+    # An DistributedItemSampler object for handling mini-batch sampling on
+    # multiple replicas.
+    ############################################################################
+    datapipe = gb.DistributedItemSampler(
+        item_set=itemset,
+        batch_size=args.batch_size,
+        drop_last=drop_last,
+        shuffle=shuffle,
+        drop_uneven_inputs=drop_uneven_inputs,
+    )
+
+    ############################################################################
+    # [Step-2]:
+    # datapipe.sample_neighbor()
+    # [Input]:
+    # 'datapipe' is either 'ItemSampler' or 'UniformNegativeSampler' depending
+    # on whether training is needed.
+    # 'graph': The network topology for sampling.
+    # 'fanouts': Number of neighbors to sample per node.
+    # [Output]:
+    # A sample_neighbor object to sample neighbors.
+    ############################################################################
+    datapipe = datapipe.sample_neighbor(graph, args.fanout)
+
+    ############################################################################
+    # [Step-3]:
+    # datapipe.fetch_feature()
+    # [Input]:
+    # 'features': The node features.
+    # 'node_feature_keys': The keys of the node features to be fetched.
+    # [Output]:
+    # A FeatureFetcher object to fetch node features.
+    ############################################################################
+    datapipe = datapipe.fetch_feature(
+        features, node_feature_keys=["feat", "label"]
```

Comment:
Does this line unnecessarily read the labels for all input nodes instead of only for the initial seed vertices?

Reply:
For now, I'm afraid so

Reply:
Is there an issue keeping track of this? I guess `fetch_feature` requires another argument for the output_nodes. Maybe output_feature_keys can be added?

Reply:
Sorry it's actually my mistake. Labels are fetched previously and only seed nodes will be fetched. Codes are updated.